pilot/pkg/model/push_context.go

```diff
@@ -95,6 +96,9 @@ type PushContext struct {
 	// ServiceAccounts contains a map of hostname and port to service accounts.
 	ServiceAccounts map[host.Name]map[int][]string `json:"-"`
 
+	// AuthNPolicies contains a map of hostname and port to authentication policy
+	AuthNPolicies map[host.Name]map[int]*Config `json:"-"`
```

Comment:
@howardjohn what happens in the two service entry with same hostname case?

Reply:
Same with above `ServiceAccounts`

Reply:
@rshriram this was the particular scenario:
```
Say I have a ServiceA port 80, and ServiceB port 90. With same hostname
The first one populates map[hostname][80]=whaterver
Then the second one comes, wipes this out with ps.AuthPolicies[svc.Hostname] = map[int]*Config{} and populates map[hostname][90]=whaterver
```
@hzxuzhonghu I pushed a separate fix for `ServiceAccounts` here https://github.com/istio/istio/pull/16769